### ISSUE TITLE
RSS search feed categories not handled for YggTorrent

### ISSUE THREAD

**Issue by gotson** [Prowlarr/Indexers#55]
I create a new issue to track this. Not sure if it should be in https://github.com/Prowlarr/Indexers instead, feel free to move it if needed.

When searching via RSS, the categories for the YGGTorrent indexer is not cascaded to the actual search.

YGGTorrent is in v1. However it seems to be working for some other v1 indexers like JPopSuki.

### YggTorrent search

`http://{server}/2/api?t=search&apikey=8c8b5c3b3af441a8a712806e44242524&cat=7000`

```
[Info] NzbSearchService: Searching indexer(s): [YGGtorrent] for Term: [], Offset: 0, Limit: 0, Categories: [7000] 
[Info] Cardigann: Adding request: https://www3.yggtorrent.nz/engine/search?category=all&name=&description=&file=&uploader=&sub_category=&do=search&order=desc&sort=publish_date 
[Info] Cardigann: Adding request: https://www3.yggtorrent.nz/engine/search?category=all&name=&page=50&description=&file=&uploader=&sub_category=&do=search&order=desc&sort=publish_date
```

### JPopSuki search 1

`http://{server}/8/api?t=search&apikey=8c8b5c3b3af441a8a712806e44242524&cat=3000`

```
[Info] NzbSearchService: Searching indexer(s): [JPopsuki] for Term: [], Offset: 0, Limit: 0, Categories: [3000] 
[Info] Cardigann: Adding request: https://jpopsuki.eu/ajax.php?filter_cat[1]=1&filter_cat[2]=1&searchtags=japanese&tags_type=0&searchstr=&order_by=s3&order_way=desc&disablegrouping=1&section=torrents
```

### JPopSuki search 2

`http://{server}/8/api?t=search&apikey=8c8b5c3b3af441a8a712806e44242524&cat=5000`

```
[Info] NzbSearchService: Searching indexer(s): [JPopsuki] for Term: [], Offset: 0, Limit: 0, Categories: [5000] 
[Info] Cardigann: Adding request: https://jpopsuki.eu/ajax.php?filter_cat[5]=1&filter_cat[6]=1&filter_cat[7]=1&searchtags=japanese&tags_type=0&searchstr=&order_by=s3&order_way=desc&disablegrouping=1&section=torrents
```

---

> sigh...
> 
> `cat`, not `cats`
> 
> also added: https://wiki.servarr.com/en/prowlarr/search#api-endpoints

I tried that before, and it works only for some indexers. For example on FileList it's working, but for YggTorrent it's not propagating to the actual indexer search:

```
[Info] NzbSearchService: Searching indexers: [YGGtorrent] for {Term: , Offset: 0, Limit: 0, Categories: [7000]} 
[Info] Cardigann: Adding request: https://www3.yggtorrent.nz/engine/search?category=all&name=&description=&file=&uploader=&sub_category=&do=search&order=desc&sort=publish_date 
[Info] Cardigann: Adding request: https://www3.yggtorrent.nz/engine/search?category=all&name=&page=50&description=&file=&uploader=&sub_category=&do=search&order=desc&sort=publish_date 
```

_Originally posted by @gotson in https://github.com/Prowlarr/Prowlarr/issues/182#issuecomment-907727370_

[AB#1788](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1788)

[AB#1793](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1793)

**Comment by gotson** [comment]
> [meta - xfered to indexers repo] @gotson seems like you have the passion and know how to do some testing with the definition and see hardcoding is still needed / if changes can be made to use dynamic cats like every other indexer and not a hardcoded config option?

Thanks for finding the historical reasons and some context, that helps knowing that the issue is in the indexer definition and not somewhere else. I will try to fix this, and if I get satisfactory results i'll submit a PR.

**Comment by gotson** [comment]
> @gotson if you're not already aware, you can test it out as a custom yml. swap the filename and id in the yml so it doesn't conflict
> 
> https://wiki.servarr.com/prowlarr/indexers#adding-a-custom-yml-definition

yes, i figured that much from the doc already, thanks!

**Comment by bakerboy448** [comment]
I'll defer to @Qstick on his thoughts for that

**Comment by gotson** [comment]
I'm also puzzled as how Jackett can handle this, given it has the same Cardigann definition 🤔

**Comment by bakerboy448** [comment]
> I'm also puzzled as how Jackett can handle this, given it has the same Cardigann definition 🤔

Pretty sure jackett would have the same issue....

I think it may just be easiest to flip the category - using the same categories available in the drop down to the user - based on the parent category sent in the request from \*arr if the keywords are empty

that'd be the simplest  route

**Comment by ilike2burnthing** [comment]
https://github.com/Jackett/Jackett/pull/14248

**Comment by ilike2burnthing** [comment]
https://github.com/Jackett/Jackett/pull/14250

Removes old category setting and the new sub-cat setting from https://github.com/Jackett/Jackett/pull/14248 earlier.

Closest thing to addressing this issue other than performing a separate search for every selected sub-category.

See info note:

> To avoid unnecessary additional requests, it's recommended to only use indexer-specific categories (>=100000) when configuring this indexer in Sonarr, Radarr, and Lidarr, or when manually searching in TV, Movies, and Audio categories.

**Comment by ilike2burnthing** [comment]
resolved in https://github.com/Prowlarr/Indexers/commit/0bbd6041f1137b425bc76918fd26e8a523852b3b

**Comment by bakerboy448** [comment]
Any fix would be related to / would need to fix https://github.com/Jackett/Jackett/issues/13697

**Comment by ilike2burnthing** [comment]
https://github.com/Jackett/Jackett/pull/14250 actually takes advantage of https://github.com/Jackett/Jackett/issues/13697 not being fixed, as we only want to use `category=all` when no categories are selected.

Short of performing individual searches for every selected subcategory, as far as I can see this issue is resolved. Or rather it was, https://github.com/Prowlarr/Indexers/pull/333 temporarily reverted the fix due to https://github.com/Prowlarr/Prowlarr/issues/1628.
